app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -358,11 +358,23 @@ class FormController(
       implicit request => implicit l => cache => _ => formModelOptics =>
         def processEditAddToList(processData: ProcessData, idx: Int, addToListId: AddToListId): Future[Result] = {
 
-          val addToListIteration =
-            processData.formModelOptics.formModelVisibilityOptics.formModel.brackets.addToListById(addToListId, idx)
+          val brackets: BracketsWithSectionNumber[Visibility] =
+            processData.formModelOptics.formModelVisibilityOptics.formModel.brackets
+          val addToListIteration = brackets.addToListById(addToListId, idx)
 
           def defaultNavigation(): (SectionNumber, FastForward) = {
-            val firstAddToListPageSN: SectionNumber = addToListIteration.firstSectionNumber
+            val firstAddToListPageSN: SectionNumber = processData.formModel.brackets
```

Comment:
We should find `firstAddToListPageSN` in view model and then look to visibility model for same or bigger section number.